src/Backend/OnboardingInsuranceAPI/Areas/Insurance/AddInsurance.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using OnboardingInsuranceAPI.ErrorHandling;
+using OnboardingInsuranceAPI.Services;
+
+namespace OnboardingInsuranceAPI.Areas.Insurance;
+
+public class AddInsurance: IHandler
+{
+    private readonly DataContext _context;
+    private readonly ILogger<AddInsurance> _logger;
+
+    public AddInsurance(DataContext context, ILogger<AddInsurance> logger)
+    {
+        _context = context;
+        _logger = logger;   
+    }
+
+    public async Task Handle(InsuranceData item)
+    {
+        if (string.IsNullOrEmpty(item.Pid))
+        {
+            _logger.LogWarning("Error, received pid is empty!");
```

Comment:
You could also put some kind of information about the insurance (some data from it - like name or something), so that you can then connect the two informations. But this is fine.